### ISSUE TITLE
Gradle 4.0 warnings about getClassesDir() when generating swagger docs with sourceSets

### ISSUE THREAD

**Issue by garyschulteog** [gigaSproule/swagger-gradle-plugin#33]
since upgrading to gradle 4.2.1 I have been seeing these warnings:

"Gradle now uses separate output directories for each JVM language, but this build assumes a single directory for all classes from a source set. This behaviour has been deprecated and is scheduled to be removed in Gradle 5.0"
  
...
 at com.benjaminsproule.swagger.gradleplugin.GenerateSwaggerDocsTask.getClasspath(GenerateSwaggerDocsTask.groovy:193)
        at com.benjaminsproule.swagger.gradleplugin.GenerateSwaggerDocsTask.processSwaggerPluginExtension(GenerateSwaggerDocsTask.groovy:64)
...

It looks like classesDir is now a collection (classesDirs), and the legacy behavior is to get the first element of the list.

**Comment by gigaSproule** [comment]
This shouldn't be happening. What version of the plugin are you using? It's meant to use `getClassesDirs` if it is available.

**Comment by garyschulteog** [comment]
classpath 'com.benjaminsproule:swagger-gradle-plugin:+' is the dependency, I am having difficulty sorting out what version that resolves to, but I presume it would be latest.  Also, the warnings do not fail the task, so this isn't a blocker for me at the moment.

**Comment by gigaSproule** [comment]
+ should mean latest (with a 24 hour cache period). If you look at https://github.com/gigaSproule/swagger-gradle-plugin/blob/58e558bc9ed6cb8d2d95b9628e479a72d4baed7f/src/main/groovy/com/benjaminsproule/swagger/gradleplugin/GenerateSwaggerDocsTask.groovy#L193, you can see it's _meant_ to use classesDirs if it is available. I'll have to take a look at it when I have time (unless you feel like having a go at it...)

**Comment by gigaSproule** [comment]
I just tried it out and I was able to see that message, but if I comment out the line `urls.add(mainSourceSetsOutput.classesDir.toURI().toURL())` it still showed it for some reason. If I commented out the whole if/else statement so it was just
```
mainSourceSetsOutput.classesDirs.each {
    urls.add(it.toURI().toURL())
}
```
it didn't show it. So it still works with older versions of gradle I'm going to have to keep it in. I'll re-visit this when it comes to 5.0
